Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -142,14 +142,82 @@
+        '''
+        """A wrapper for requests lib to send our requests and handle requests and responses better.
+
+            :type method: ``str``
+            :param method: The HTTP method, for example: GET, POST, and so on.
+
+            :type url_suffix: ``str``
+            :param url_suffix: The API endpoint.
+            
+            :type full_url: ``str``
+            :param full_url:
+                Bypasses the use of self._base_url + url_suffix. This is useful if you need to
+                make a request to an address outside of the scope of the integration
+                API.
+
+            :type headers: ``dict``
+            :param headers: Headers to send in the request. If None, will use self._headers.
+            
+            :type params: ``dict``
+            :param params: URL parameters to specify the query.
+
+            :type data: ``dict``
+            :param data: The data to send in a 'POST' request.
+            
+            :type raise_on_status ``bool``
+                :param raise_on_status: Similar meaning to ``raise_on_redirect``:
+                    whether we should raise an exception, or return a response,
+                    if status falls in ``status_forcelist`` range and retries have
+                    been exhausted.
+            
+            
+            :type timeout: ``float`` or ``tuple``
+            :param timeout:
+                The amount of time (in seconds) that a request will wait for a client to
+                establish a connection to a remote machine before a timeout occurs.
+                can be only float (Connection Timeout) or a tuple (Connection Timeout, Read Timeout).
+        '''
+        if not FORWARD_USER_RUN_RBAC:
+            return BaseClient._http_request(self, method=method, url_suffix=url_suffix, full_url=full_url, headers=headers,
+                                            json_data=json_data,params=params, data=data, timeout=timeout,
+                                            raise_on_status=raise_on_status)
+        try:
+            # Replace params if supplied
+            address = urljoin(self._base_url, url_suffix)
+            headers = headers if headers else self._headers
+            data = json.dumps(json_data) if json_data else data
+            address = full_url if full_url else urljoin(self._base_url, url_suffix)
+            response = demisto._apiCall(
+                        method=method,
+                        path=address,
+                        data=data,
+                        headers=self._headers,
+                        timeout=timeout
+                    )
+            if ok_codes and response.get('status') not in ok_codes:
+                    self._handle_error(error_handler, response, with_metrics)
+            return json.loads(response['data'])
+        except Exception as exception:
+            raise DemistoException(exception)
```

Comment:
I'd remove the try-except, now you just replace the exception type to DemistoException, better keep the original